src/main/java/hudson/plugins/git/UserRemoteConfig.java

```diff
@@ -127,7 +129,7 @@ public FormValidation doCheckCredentialsId(@AncestorInPath Item project,
             return FormValidation.ok();
         }
 
-        public FormValidation doCheckUrl(@AncestorInPath Item project,
+        public FormValidation doCheckUrl(@AncestorInPath Job project,
```

Comment:
Yeah, it's incomplete but we're just doing a UI-side validation here, so it doesn't need to be 100% exhaustive --  and as you say, we don't have any cases where GitSCM is used with an Item that's not a Job (yet, AFAIK).